lang_java/tree_sitter/CONTRIBUTING_TREE_SITTER.md

```diff
@@ -0,0 +1,63 @@
+# Contributing
+
+## Overview
+
+This directory houses the files used for converting `tree-sitter` grammar to `pfff` grammar. 
+
+## Goal 
+
+We want to have the following components working for a fully-functional polyglot parser to integrate into sgrep:
+- `parse_java_with_external_program.ml`: takes a `.java` file and returns a `.json` file, then runs a parser over the `.json` file to construct a `Tree_sitter_ast_java.ast` type object 
+- `tree_sitter_ast_java.ml`: defines a grammar based on the `grammar.js`/`grammar.json` from `tree-sitter-java` 
+- `tree_sitter_to_ast_java.ml`: (does not exist yet) converts the `Tree_sitter_ast_java.ast` type object to a `Ast_java.ast` type object
```

Comment:
I would a line for each of these to explain what they are supposed to do from a high level (not necessarily functionality)

Reply:
A system diagram would be even better!

Reply:
Ok I added that!